lib/components/section_update_handler.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:brunstadtv_app/graphql/queries/sections.graphql.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../graphql/client.dart';
+import '../graphql/queries/page.graphql.dart';
+import '../helpers/event_bus.dart';
+import '../models/events/my_list_changed.dart';
+import '../providers/section_updates.dart';
+import '../helpers/extensions.dart';
+import 'section.dart';
+
+class SectionUpdateHandler extends HookConsumerWidget {
+  const SectionUpdateHandler({
+    super.key,
+    required this.section,
+    this.extraItems,
+  });
+
+  final Fragment$Section section;
+  final List<Fragment$ItemSectionItem>? extraItems;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final updatedSection = useState(section);
+    final isMyList = section.asOrNull<Fragment$ItemSection>()?.metadata?.myList == true;
+
+    useEffect(() {
+      updatedSection.value = section;
+      return null;
+    }, [section]);
+
+    void refreshSection() {
+      ref
+          .watch(gqlClientProvider)
+          .query$GetSection(
+            Options$Query$GetSection(
+              variables: Variables$Query$GetSection(
+                id: section.id,
+                timestamp: DateTime.now().toIso8601String(),
+              ),
+            ),
+          )
+          .then(
+        (response) {
+          final section = response.parsedData?.section;
+          if (section != null) {
+            updatedSection.value = section;
+          }
+        },
+      );
+    }
+
+    ref.listen(sectionUpdatesProvider(section.id), (prev, next) => refreshSection());
+
+    if (isMyList) {
+      globalEventBus.on<MyListChangedEvent>().listen((event) => refreshSection());
```

Comment:
this is going to create a new listener every build(). You need to use useEffect() to only register it when "isMyList" changes, and the return value in useEffect should kill the listener
so psuedocode (dont know if its correct syntax):

```dart
useEffect(() {
  if (!isInMyList) return null;
  final listener = globalEventBus.on<MyListChangedEvent>().listen((event) => refreshSection());
  return listener.cancel;
}, [isInMyList]);
```

Reply:
Oh right.